### PR TITLE
mypy_extensions: add recipe

### PR DESCRIPTION
**Pull request by nehaljwani**

### CODE REVIEW

File: recipes/mypy_extensions/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  number: 0
+  script: python setup.py install
+
+requirements:
+  build:
+    - python
+
+  run:
+    - python
+    - typing  # [py<35]
```
Comment (jakirkham) [inline recipes/mypy_extensions/meta.yaml]: Does this require `mypy` as well?

Comment (jakirkham) [inline recipes/mypy_extensions/meta.yaml]: Seems they don't list this requirement in their `setup.py`. Would raise an issue to get it added.

Comment (nehaljwani) [reply recipes/mypy_extensions/meta.yaml]: Seeing the source code at https://github.com/python/mypy/blob/master/extensions/mypy_extensions.py doesn't have any reference to mypy.

Comment (nehaljwani) [reply recipes/mypy_extensions/meta.yaml]: Done. https://github.com/python/mypy/issues/4264

**Comment by mwcraig** [comment]
Thanks for opening the upstream issue/PR, @nehaljwani, merging.

**Pull request merged**